duo_client/admin.py

```diff
@@ -2653,6 +2647,80 @@ def activate_admin(self, email,
                                       params)
         return response
 
+    def get_external_password_mgmt_statuses(self, limit=None, offset=0):
+        """
+        Returns a paged list of administrators indicating whether they
+        have been configured for external password management.
+        Args:
+            limit: The max number of admins to fetch at once. Default None
+            offset: If a limit is passed, the offset to start retrieval.
+                    Default 0
+
+        Returns a list of administrators' external password management
+            statuses.  See the adminapi docs.
+
+        Raises RuntimeError on error.
+        """
+
+        (limit, offset) = self.normalize_paging_args(limit, offset)
+        if limit:
+            return self.json_api_call(
+                'GET',
+                '/admin/v1/admins/password_mgmt',
+                {'limit': limit, 'offset': offset}
+            )
+
+        iterator = self.json_paging_api_call(
+            'GET', '/admin/v1/admins/password_mgmt', {})
+
+        return list(iterator)
+
+    def get_external_password_mgmt_status_for_admin(self, admin_id):
+        """
+        Returns the external password management status for an admin
+
+        admin_id - The id of the admin.
+
+        Returns an external password management status. See the
+        adminapi docs.
+
+        Raises RuntimeError on error.
+        """
+
+        admin_id = six.moves.urllib.parse.quote_plus(str(admin_id))
+        path = '/admin/v1/admins/{}/password_mgmt'.format(admin_id)
+        response = self.json_api_call('GET', path, {})
+        return response
+
+    def update_admin_password_mgmt_status(
+        self, admin_id, has_external_password_mgmt=None,
+        password=None):
+        """
+        Enable or disable an admin for external password management,
+        and optionally set the password for an admin
+
+        admin_id - The id of the admin.
+        has_external_password_mgmt - whether or not this admin's password
+            is managed via API.
+        password - New password for the admin.
+
+        Returns an external password management status. See the
+        adminapi docs.
+
+        Raises RuntimeError on error.
+        """
+
+        params = {}
+        if password is not None:
+            params['password'] = password
+        if has_external_password_mgmt is not None:
```

Comment:
are these args mutually exclusive? If so, maybe we should raise an Exception if both are specified? or maybe we should let the Duo API return that error instead of enforcing it in the client?

Reply:
These are not mutually exclusive!

However, if the value of `has_external_password_mgmt` is False and `password` is provided, then the call will fail.  I figured we can just let the API return the error in this case.